### ISSUE TITLE
Facing issues with undefined symbols

### ISSUE THREAD

**Issue by tinder-maxwellelliott** [google/google-toolbox-for-mac#275]
Hello Y'all,

I am trying to build Google Toolbox for Mac using Bazel and [PodToBuild](https://github.com/pinterest/PodToBUILD). Everything seems to compile fine, but when I go to link my binary to this library I get these errors:

```terminal
undef: _kGTMRegexPatternErrorErrorString
undef: _kGTMRegexPatternErrorPattern
undef: _kGTMRegexErrorDomain
Undefined symbols for architecture x86_64:
  "_kGTMRegexPatternErrorErrorString", referenced from:
      -[GTMRegex initWithPattern:options:withError:] in libRegex.a(GTMRegex.o)
  "_kGTMRegexPatternErrorPattern", referenced from:
      -[GTMRegex initWithPattern:options:withError:] in libRegex.a(GTMRegex.o)
  "_kGTMRegexErrorDomain", referenced from:
      -[GTMRegex initWithPattern:options:withError:] in libRegex.a(GTMRegex.o)
ld: symbol(s) not found for architecture x86_64
```

Is there some sort of additional flag I need to pass to compile this correctly? From what I can see reading the PodSpec, there is no custom logic I can see or flags passed?

For what it's worth, the `__EXTERN` style used in Foundation/GTMRegex.h differs from the rest of the codebase.

Any help is greatly appreciated.

**Comment by tinder-maxwellelliott** [comment]
Also when running `otool` it looks like x86_64 is the supported architecture

```
otool -hv -arch all bazel-out/apl-ios_x86_64-fastbuild/bin/external/GoogleToolboxForMac/libRegex.a
Archive : bazel-out/apl-ios_x86_64-fastbuild/bin/external/GoogleToolboxForMac/libRegex.a
bazel-out/apl-ios_x86_64-fastbuild/bin/external/GoogleToolboxForMac/libRegex.a(GTMRegex.o):
Mach header
      magic  cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
MH_MAGIC_64   X86_64        ALL  0x00      OBJECT     5       2528 SUBSECTIONS_VIA_SYMBOLS
```

**Comment by thomasvl** [comment]
Likely specific to PodToBuild, I don't think any of us have really looked at it, but it likely isn't compiling the files individual so the guards between the headers/sources works to create the definitions.

**Comment by tinder-maxwellelliott** [comment]
Here is a link to the generated BUILD file from PodToBuild https://gist.github.com/tinder-maxwellelliott/7abb221fc267361237a42147be1b0ca8

It might be time to finally create a BUILD file as mentioned in #266

**Comment by thomasvl** [comment]
Given the complexity in that generated BUILD, I'm not that surprised things don't work out, looks like it might be doing some custom generation of pchs, etc.

**Comment by thomasvl** [comment]
This should be fixed by #277
